rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php

```diff
@@ -144,6 +155,13 @@ public function provideMinPhpVersion(): int
         return PhpVersionFeature::CLASSNAME_CONSTANT;
     }
 
+    private function fillIsUnderClassConstAttribute(ClassConst $classConst): void
+    {
+        $nodeTraverser = new NodeTraverser();
+        $nodeTraverser->addVisitor(new ClassConstStringValueNodeVisitor());
+        $nodeTraverser->traverse([$classConst]);
+    }
```

Comment:
The traversing and attributes should happen in one place before the rectors start running, so we clear separation of atrtibutes and node processing.

Reply:
This pattern should be avoided in the first place, so it doesn't spread like parent/next/previous nodes. 

What is the issue here anyway? Could you add a test case to show what is failing?

Reply:
Using inside `beforeTraverse()` will look too up too early on root Stmt, so that's no go. If it going to be used in multiple rules, we can move to under PHPStan namespace:

https://github.com/rectorphp/rector-src/tree/8b8ac6e2d9ea276664ed2ff181799a4ede94149a/packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor

Reply:
`STOP_TRAVERSAL` is only allowed when inside `SimpleCallableNodeTraverser` which traversed in separate process, otherwise, next Rector rule will just stopped, like what we see in dowgrade rules.

That means, upgrading from php 5.6 never executed after it, that's why fill attribute is needed, the use case is like this:

```php
 final class SkipInArrayConstant
 {
     const SKIP_TYPES = [
        'Rector\Tests\Php55\Rector\String_\StringClassNameToClassConstantRector\Source\SomeUser'
     ];

     public function run()
     {
             echo $undefinedVariableHere;
      }
 }
```

when run against PHP 5.6 add default undefined variable, that's not executed anymore.

Reply:
Here the failing test case PR:

- https://github.com/rectorphp/rector-src/pull/4185